### PR TITLE
Update and evaluate relay docs structure

### PR DESCRIPTION
**Pull request by PeloWriter**
This moves the Relay docs into the Security content, as its primary audience is SecOps personnel. In addition, this PR includes edited content to flesh out some of the value that Relay brings to our customers as well as smoothing and clarifying some areas. 

One technical note: the Configuration Options page discusses a mode of `capture` , but the Relay Modes page does not mention this.

Screen Shot 2020-09-16 at 10 54 22 AM

Note: no redirects are required, as this content isn't yet published.

### CODE REVIEW

File: src/docs/meta/relay/index.mdx [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+draft: true
+noindex: true
+sidebar_order: 1
+---
+
+Sentry Relay offers enterprise-grade data security by providing a standalone service that acts as a middle layer between your application and sentry.io.
+
+Relay is specifically designed to:
+- Scrub personal information in a central place prior to sending it to Sentry, adding to the places Sentry already scrubs PII
+- Improve event response time
```
Comment (jan-auer) [inline src/docs/meta/relay/index.mdx:13]: Let's add to this:

```suggestion
- Improve event response time in regions with low bandwidth or limited connectivity
```
```suggestion
- Improve event response time
```

File: src/docs/meta/relay/modes.mdx [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+title: Relay Modes
+description: "Relay can operate in different modes depending on your organization's needs."
+sidebar_order: 2
+---
+
+Relay can operate in one of several major modes, and it is critical to understand the modes prior if you are configuring the Relay server rather than using the default settings.
+
+The mode is stored in the configuration file, which contains the `relay.mode` field. This field specifies the mode in which Relay will run: `managed`, `static`, or `proxy`. The Relay mode controls the way Relay obtains project settings for events.
+
+In Sentry, event processing is configured according to both project and organization settings. Some settings, such as privacy controls, are set at the organization level, then inherited by all projects in that organization; other settings are specified per project. For Relay, events are processed according to the inherited project settings to which the event belongs.
```
Comment (jan-auer) [inline src/docs/meta/relay/modes.mdx:11]:
```suggestion
In Sentry, event processing is configured according to both project and organization settings. Some settings, such as privacy controls, are set at the organization level, then inherited by all projects in that organization; other settings are specified per project. For Relay, events are processed according to the inherited project settings to which the event is sent.
```

File: src/docs/meta/relay/modes.mdx [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+
+Configuration for Relay is refreshed in regular intervals. See [Configuration Options] regarding configuration of intervals, timeouts, and retries.
```
Comment (jan-auer) [inline src/docs/meta/relay/modes.mdx:13]: This was asked by customers multiple times, so I suggest to add:

```suggestion
Configuration for Relay is refreshed in regular intervals by polling Sentry. Sentry does not need line-of-sight to your Relays. See [Configuration Options] regarding configuration of intervals, timeouts, and retries.
```
```suggestion
Configuration for Relay is refreshed in regular intervals. See [Configuration Options] regarding configuration of intervals, timeouts, and retries.
```

File: src/docs/meta/relay/modes.mdx [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+
+## Managed Mode
+
+Managed is Relay's default mode. Because settings are obtained from Sentry, authentication is required in this mode. If authentication fails, no events will be accepted by Relay.
+
+As Relay receives events from your applications, it will request project
+settings from Sentry to process the events. If Sentry is unable to provide
+the settings for a particular project, all messages for that project will
```
Comment (jan-auer) [inline src/docs/meta/relay/modes.mdx:21]:
```suggestion
the settings for a particular project, all data for that project will
```
```suggestion
the settings for a particular project, all messages for that project will
```

File: src/docs/meta/relay/options.mdx [hunk 5]
```diff
@@ -109,7 +110,7 @@ Set various network-related settings.
 
 ## Caching
 
-Fine-tune caching of project state.
+Thesee setting fine tune caching of project state.
```
Comment (jan-auer) [inline src/docs/meta/relay/options.mdx:113]:
```suggestion
These settings fine tune caching of project states.
```

File: src/docs/meta/relay/options.mdx [hunk 6]
```diff
@@ -179,7 +180,7 @@ Fine-tune caching of project state.
 
 ## Size Limits
 
-Controls various HTTP-related limits. All values are either integers or are
+These setting control HTTP-related limits. All values are either integers, or are
```
Comment (jan-auer) [inline src/docs/meta/relay/options.mdx:183]:
```suggestion
These settings control HTTP-related limits. All values are either integers, or are
```

File: src/docs/meta/relay/projects.mdx [hunk 7]
```diff
@@ -0,0 +1,109 @@
...
+  ],
+  "config": {
+    "allowedDomains": ["*"]
+  }
+}
+```
+
+<Alert level="warning" title="Public Key">
+
+The public key (`<DSN_KEY>`) is the key of the project's DSN and is unrelated toh the Relay public key used for Relay registration.
```
Comment (jan-auer) [inline src/docs/meta/relay/projects.mdx:39]:
```suggestion
The public key (`<DSN_KEY>`) is the key of the project's DSN and is unrelated to the Relay public key used for Relay registration.
```

**Review by PeloWriter — COMMENTED** [review]
I will merge as as soon as the checks pass.

**Pull request merged**